packages/app/src/components/CardInfoLifecycle.tsx

```diff
@@ -1,31 +1,192 @@
-import { InfoButton, InfoSheet } from '@package/ui'
-import React from 'react'
+import { Button, HeroIcons, InfoButton, InfoSheet, Stack, useToastController } from '@package/ui'
+import type { StatusVariant } from '@package/ui/src/utils/variants'
+import { formatDate, formatDaysString, getDaysUntil } from '@package/utils/src'
+import React, { useEffect, useMemo } from 'react'
 import { useState } from 'react'
 import { useHaptics } from '../hooks/useHaptics'
 
-export function CardInfoLifecycle() {
+// Expired requires a different flow (see component below)
+type BaseLifeCycle = 'active' | 'revoked' | 'batch'
+
+type LifeCycleContent = {
+  variant: StatusVariant
+  title: string
+  description: string
+  sheetDescription: string
+}
+
+const cardInfoLifecycleVariant: Record<BaseLifeCycle, LifeCycleContent> = {
+  active: {
+    variant: 'positive',
+    title: 'Card is active',
+    description: 'No actions required',
+    sheetDescription: 'Your credentials may expire or require an active internet connection to validate.',
+  },
+  revoked: {
+    variant: 'danger',
+    title: 'Card revoked',
+    description: 'Card not usable anymore',
+    sheetDescription:
+      'The issuer has revoked this card and it can not be used anymore. Contact the issuer for more information.',
+  },
+  // We can hardcode this to the rules for the PID credential as this will be the only of this type for now.
+  batch: {
+    variant: 'warning',
+    title: 'Limited card usage',
+    description: 'verifications left',
+    sheetDescription:
+      'This card requires periodic validation using an internet connection. When usage is low you will be notified.',
+  },
+}
+
+interface CardInfoLifecycleProps {
+  validUntil?: Date
+  validFrom?: Date
+  isRevoked?: boolean
+  batchLeft?: number
+}
+
+export function CardInfoLifecycle({ validUntil, validFrom, isRevoked, batchLeft }: CardInfoLifecycleProps) {
+  const toast = useToastController()
+  const [isOpen, setIsOpen] = useState(false)
+  const { withHaptics } = useHaptics()
+
+  const state = useMemo(() => {
+    if (isRevoked) return 'revoked'
+    if (batchLeft) return 'batch'
+
+    return 'active'
+  }, [isRevoked, batchLeft])
+
+  const onPress = withHaptics(() => setIsOpen(!isOpen))
+
+  const onPressValidate = withHaptics(() => {
+    // Implement navigation to the setup eID card flow.
+    toast.show('Coming soon', { customData: { preset: 'warning' } })
+  })
+
+  if (validUntil || validFrom) {
+    return <CardInfoLimitedByDate validUntil={validUntil} validFrom={validFrom} />
+  }
+
+  return (
+    <>
+      {batchLeft && batchLeft <= 5 && (
+        <Stack pb="$4">
+          <Button.Solid bg="$grey-50" bw="$0.5" borderColor="$grey-200" color="$grey-900" onPress={onPressValidate}>
+            Refresh card <HeroIcons.ArrowPath ml="$-2" size={20} color="$grey-700" />
```

Comment:
So i asked about refresh and they recommended me to use the C" issuance flow which does support refresh tokens. 

So the flow would be:
- authentication using eid card first time. Get batch and refresh token
- use refresh token in background to retrieve new token and get new batch
- if refresh token also expires we need to trigger eID card flow. 


But this will be a lot better already. So i think the "only x left in batch" should only be shown if we werent't able to automatically get new ones and re-authentication is required. But that's really a fallback.

We have the ReceivePidUseCaseFlow. Maybe if you can add these placeholder methods for now I'll implement these later:
- tryRetrieveBatchUsingRefreshToken


Then if that method fails because of unauthorized we need to trigger eID flow, or at least show to the user that they only have X verifications left and they need to re-authenticate.